### PR TITLE
Do not bail on SHOW_CONFIG_FOR matches

### LINKED ISSUE #12425: Reconsider bailing out with `BABEL_SHOW_CONFIG_FOR`
## Bug Report

**Current behavior**

When using `BABEL_SHOW_CONFIG_FOR`, the file for which Babel is logging the config is considered as ignored and it's not compiled/handled: https://github.com/babel/babel/blob/cb30c1dd54fd09f05045b5094ff9d038604768b3/packages/babel-core/src/config/config-chain.js#L266

However, all the other files are still compiled. This means that if my build script is `babel src -d out`, if I have two files (`src/main.js` and `src/dep.js`) and I run
```
BABEL_SHOW_CONFIG_FOR=./src/main.js npm run build
```

I will get an `out` folder containing only the `dep.js` file.

If you want to see this behavior, you can clone https://github.com/nicolo-ribaudo/babel-bug-show-config-behavior and run `BABEL_SHOW_CONFIG_FOR=./src/main.js npm run build`.

**Expected behavior**

Either Babel should compile everything, or nothing.

I think the best thing to do would be to still compile the file, so that `BABEL_SHOW_CONFIG_FOR` only has effect on logging and on nothing else. However, I would be open to ignoring everything.

**Additional context**

Thanks @laurieontech for the feedback!

### PR DESCRIPTION
**Pull request by JLHwung**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | Fixes #12425
| Patch: Bug Fix?          |
| Major: Breaking Change?  |
| Minor: New Feature?      |
| Tests Added + Pass?      | Yes
| Documentation PR Link    | 
| Any Dependency Changes?  |
| License                  | MIT

I will mark this PR ready when the Windows CI is green. 

The test fixtures are updated because now Babel will actually resolve presets and transform the code. It turns out the complex config in out test cases is not a valid config.

### CODE REVIEW

File: packages/babel-cli/test/fixtures/babel/env - SHOW_CONFIG_FOR/stdout.txt [hunk 1]
```diff
@@ -125,3 +124,4 @@ programmatic options from @babel/cli
   },
   "filename": "./src/index.js"
 }
+"use strict";
```
Comment (nicolo-ribaudo) [inline packages/babel-cli/test/fixtures/babel/env - SHOW_CONFIG_FOR/stdout.txt:127]: Maybe we should print a separator between the code and the config? Something like
```

================================================================================

```

File: packages/babel-cli/test/fixtures/babel/env - SHOW_CONFIG_FOR/in-files/my-extended.js [hunk 2]
```diff
@@ -11,7 +11,6 @@ module.exports = {
             "@foo/babel-preset-1",
             [
               {
-                name: "@foo/inline-babel-preset-1",
```
Comment (nicolo-ribaudo) [inline packages/babel-cli/test/fixtures/babel/env - SHOW_CONFIG_FOR/in-files/my-extended.js:14]: Q: What is this change needed for?

Comment (JLHwung) [reply packages/babel-cli/test/fixtures/babel/env - SHOW_CONFIG_FOR/in-files/my-extended.js:14]: Babel will throw `name` is not a valid option. I didn't know `name` is forbidden on `preset` until now. 🤦‍♂️

File: packages/babel-cli/test/fixtures/babel/env - SHOW_CONFIG_FOR/stdout.txt (hunk 1 above)
Comment (JLHwung) [reply packages/babel-cli/test/fixtures/babel/env - SHOW_CONFIG_FOR/stdout.txt:127]: I don't think the debug output should be directed to `stderr`. 

For example `curl -v http://example.com/example.avif > example.avif` will also break, unless we specify `-O example.avif`. Similary in this case, consider use `--out-file example.js` instead.

**Review by nicolo-ribaudo — APPROVED** [review]

**Review by hzoo — APPROVED** [review]
lgtm

aside: looks like the only place we have docs on this is in https://babeljs.io/docs/en/configuration#print-effective-configs? Maybe should link to it in the options/config page as well https://babeljs.io/docs/en/config-files?
Or also suggest running when encountering certain kinds of errors

**Comment by nicolo-ribaudo** [comment]
A good idea would be adding it to the issue template :+1:
So that people not only can tell us what config they _think_ they have, but also what Babel sees.
